x-pack/plugins/ml/public/application/components/model_snapshots/revert_model_snapshot_flyout/revert_model_snapshot_flyout.tsx

```diff
@@ -0,0 +1,396 @@
+
+  useEffect(() => {
+    createChartData();
+  }, [currentSnapshot]);
+
+  useEffect(() => {
+    const invalid = calendarEvents.some(
+      (c) => c.description === '' || c.end === null || c.start === null
+    );
+    setCalendarEventsValid(invalid === false);
+
+    // a bug in elastic charts selection can
+    // cause duplicate selected areas to be added
+    // dedupe the calendars based on start and end times
+    const calMap = new Map(
+      calendarEvents.map((c) => [`${c.start?.valueOf()}${c.end?.valueOf()}`, c])
+    );
+    const dedupedCalendarEvents = [...calMap.values()];
+
+    if (dedupedCalendarEvents.length < calendarEvents.length) {
+      // deduped list is shorter, we must have removed something.
+      setCalendarEvents(dedupedCalendarEvents);
+    }
+  }, [calendarEvents]);
+
+  async function createChartData() {
+    const bucketSpanMs = parseInterval(job.analysis_config.bucket_span)!.asMilliseconds();
+    const eventRate = await loadEventRateForJob(job, bucketSpanMs, 100);
+    const anomalyData = await loadAnomalyDataForJob(job, bucketSpanMs, 100);
+    setEventRateData(eventRate);
+    if (anomalyData[0] !== undefined) {
+      setAnomalies(anomalyData[0]);
+    }
+    setChartReady(true);
+  }
+
+  function closeWithReload() {
+    closeFlyout(true);
+  }
+  function closeWithoutReload() {
+    closeFlyout(false);
+  }
+
+  function showRevertModal() {
+    setRevertModalVisible(true);
+  }
+  function hideRevertModal() {
+    setRevertModalVisible(false);
+  }
+
+  async function applyRevert() {
+    setApplying(true);
+    const end =
+      replay && runInRealTime === false ? job.data_counts.latest_record_timestamp : undefined;
+    try {
+      const events = replay
+        ? calendarEvents.filter(filterIncompleteEvents).map((c) => ({
+            start: c.start!.valueOf(),
+            end: c.end!.valueOf(),
+            description: c.description,
+          }))
+        : undefined;
+
+      await ml.jobs.revertModelSnapshot(
+        job.job_id,
+        currentSnapshot.snapshot_id,
+        replay,
+        end,
+        events
+      );
+      hideRevertModal();
+      closeWithReload();
+    } catch (error) {
+      setApplying(false);
+      toasts.addError(new Error(error.body.message), {
+        title: i18n.translate('xpack.ml.revertModelSnapshotFlyout.revertErrorTitle', {
+          defaultMessage: 'Model snapshot revert failed',
+        }),
+      });
+    }
+  }
+
+  function onSnapshotChange(ssId: string) {
+    const ss = snapshots.find((s) => s.snapshot_id === ssId);
+    if (ss !== undefined) {
+      setCurrentSnapshot(ss);
+    }
+  }
+
+  return (
+    <>
+      <EuiFlyout onClose={closeWithoutReload} hideCloseButton size="m">
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="s">
+            <h5>
+              <FormattedMessage
+                id="xpack.ml.newJob.wizard.revertModelSnapshotFlyout.title"
+                defaultMessage="Revert snapshot {ssId}"
```

Comment:
As before, I wonder if `Revert to snapshot <ID>` would read better, or even `Revert to saved snapshot <id>` or `Revert to saved model snapshot <ID>` !?